includes/class-wc-stripe-helper.php

```diff
@@ -432,6 +432,39 @@ public static function get_order_by_charge_id( $charge_id ) {
 		return false;
 	}
 
+	/**
+	 * Gets the order by Stripe refund ID.
+	 *
+	 * @since x.x.x
+	 * @param string $refund_id
+	 */
+	public static function get_order_by_refund_id( $refund_id ) {
+		global $wpdb;
+
+		if ( class_exists( 'Automattic\WooCommerce\Utilities\OrderUtil' ) && OrderUtil::custom_orders_table_usage_is_enabled() ) {
+			$orders   = wc_get_orders(
+				[
+					'limit'          => 1,
+					'meta_query' => [
+						[
+							'key'   => '_stripe_refund_id',
+							'value' => $refund_id,
+						],
+					],
+				]
+			);
+			$order_id = current( $orders ) ? current( $orders )->get_id() : false;
+		} else {
+			$order_id = $wpdb->get_var( $wpdb->prepare( "SELECT DISTINCT ID FROM $wpdb->posts as posts LEFT JOIN $wpdb->postmeta as meta ON posts.ID = meta.post_id WHERE meta.meta_value = %s AND meta.meta_key = %s", $refund_id, '_stripe_refund_id' ) );
```

Comment:
`wc_get_orders()` seems to work here even with HPOS enabled. I assume you reused it from `get_order_by_charge_id()`. It might be better to use `wc_get_orders()` only if it works in both cases.

Reply:
If you want to test yourself, to enable HPOS you can go to WooCommerce -> Settings -> Advanced -> "Enable the high performance order storage feature"

Reply:
With HPOS disabled, unfortunately the meta_query is just ignored by `wc_get_orders()`. 

I did check this and I thought you were right for a moment, because it seems to return the correct order. But, if you test with the limit removed you can see actually all orders are being returned in date order, and naturally in testing it was normally the most recent order being tested with.

This implementation is consistent with `get_order_by_intent_id` and `get_order_by_setup_intent_id`, which would presumably have the same problem without the custom queries. `get_order_by_charge_id` is a little different because it doesn't need a meta query.

Reply:
I see, it's always returning the last order (refund). Which is a coincidence because the last refund is not necessarily the one we are handling on the incoming Webhook.